### PR TITLE
Documents consumer group lag

### PR DESCRIPTION
**Pull request by itamarhaber**
Upstream ref: https://github.com/redis/redis/pull/9127
update commands .json from unstable (pre 7.0 RC2)

### CODE REVIEW

File: wordlist [hunk 1]
```diff
@@ -429,6 +429,7 @@ virtualized
 vmstat
 vtype
 wildcards
+xdel
```
Comment (zuiderkwast) [inline wordlist:432]: Is this really neeed? How come XADD and the others are not in the wordlist?

File: commands/xinfo-groups.md [hunk 2]
```diff
@@ -6,29 +6,68 @@ By default, only the following information is provided for each of the groups:
...
+Redis reports the lag of a consumer group by keeping two offsets: one for the stream and another one for the consumer group.
+The lag is the difference between these two offsets.
+
+The stream's offset (the `last-offset` field of the `XINFO STREAM` command) is incremented by one with every `XADD` and counts all of the entries added to the stream during its lifetime.
+
+The consumer group's offset, `last-delivered-offset`, is the offset of the entry referenced by the group's `last-delivered-id`.
+
+There are two special cases in which this mechanism is unable to report the lag:
+
+1. A consumer group is created or set with an arbitrary last delivered ID (the `XGROUP CREATE` and `XGROUP SETID` commands, respectively). An arbitrary ID is any ID that isn't the ID of the stream's first entry, its last entry or the zero ("0-0") ID.
```
Comment (zuiderkwast) [inline commands/xinfo-groups.md:29]:
```suggestion
1. A consumer group is created or set with an arbitrary last delivered ID (the `XGROUP CREATE` and `XGROUP SETID` commands, respectively).
   An arbitrary ID is any ID that isn't the ID of the stream's first entry, its last entry or the zero ("0-0") ID.
```

File: commands/xinfo-groups.md [hunk 3]
```diff
@@ -6,29 +6,68 @@ By default, only the following information is provided for each of the groups:
...
+2. One or more entries between the group's `last-delivered-id` and the stream's `last-generated-id` were deleted (with `XDEL` or a trimming operation).
+
+In both cases, the group's offset is considered invalid, and the returned value is set to NULL to signal that the lag isn't currently available.
+
+However, the lag's unavailability is only temporary and resolves itself during regular operation.
```
Comment (zuiderkwast) [inline commands/xinfo-groups.md:34]: Unavailability is an unusual word. It's extra unusual that it resolves itself. :-) Maybe this sentence can be made a little lighter? Ideas:

*  "the lag's unavailability" => "the unavailability of the lag" or "absence of the lag" or "the lag is only temporarily missing"
* "resolves itself" => "becomes available again automatically"
* Split into two sentences

**Review by zuiderkwast — COMMENTED** [review]
Looks good!

I didn't proof read the examples, assuming they're generated.

Do we need to wait for and/or coordinate with redis/redis#9656 regarding commands.json?

File: commands/xinfo-groups.md (hunk 3 above)
Comment (itamarhaber) [reply commands/xinfo-groups.md:34]: Resolved.

File: wordlist (hunk 1 above)
Comment (itamarhaber) [reply wordlist:432]: No longer needed - it was part of xdel-max-id which wasn't quoted, but was renamed anyway.

**Comment by oranagra** [comment]
@itamarhaber what the hell happened to commands.json in this PR?

**Comment by oranagra** [comment]
ohh, this PR is based on a very old version of the master branch.
but you've re-genrated the commands.json, so it looks like it completely changes it.
i'll merge the master into this one and re-generate commands.json again.

**Review by oranagra — APPROVED** [review]

**Pull request merged**